core/src/stages/connect.cpp

```diff
@@ -132,6 +134,51 @@ bool Connect::compatible(const InterfaceState& from_state, const InterfaceState&
 	return true;
 }
 
+bool Connect::validateEndTrajectoryDeviation(const moveit::core::JointModelGroup* jmg,
+                                             const robot_trajectory::RobotTrajectoryPtr trajectory,
+                                             const moveit::core::RobotState& goal_state, std::string& comment) {
+	const std::size_t waypoint_count = trajectory->getWayPointCount();
+
+	if (!waypoint_count) {
+		comment = "Cannot validate end trajectory deviation with an empty trajectory";
+		return false;
+	}
+
+	for (const moveit::core::JointModel* jm : jmg->getJointModels()) {
+		const unsigned int num = jm->getVariableCount();
+
+		if (!num)
+			continue;
+
+		Eigen::Map<const Eigen::VectorXd> positions_goal(goal_state.getJointPositions(jm), num);
+		Eigen::Map<const Eigen::VectorXd> positions_last_waypoint(
+		    trajectory->getWayPoint(waypoint_count - 1).getJointPositions(jm), num);
+
+		double min_distance;
+		for (std::size_t i = 0; i < num; ++i) {
+			if (jm->getType() == robot_state::JointModel::REVOLUTE)
+				min_distance = angles::shortest_angular_distance(positions_last_waypoint[i], positions_goal[i]);
+			else
+				min_distance = positions_last_waypoint[i], positions_goal[i];
```

Comment:
I guess you want to compute the joint difference:
```suggestion
				min_distance = std::abs(positions_last_waypoint[i] -  positions_goal[i]);
```